runner/utils/driver/docker/docker.go

```diff
@@ -0,0 +1,252 @@
+				if err := d.client.ContainerKill(ctx, containerName, "SIGKILL"); err != nil {
+					logrus.Errorf("error deleting service in docker unable to kill container %s got error message - %v", containerName, err)
+					return err
+				}
+			}
+
+			// remove the container from host machine
+			if err := d.client.ContainerRemove(ctx, containerName, types.ContainerRemoveOptions{}); err != nil {
+				logrus.Errorf("error deleting service in docker unable to remove container %s got error message - %v", containerName, err)
+				return err
+			}
+		}
+	}
+
+	// handle gracefully if no containers found for specified serviceId
+	return nil
+}
+
+// GetService gets the specified service info from docker container
+func (d *docker) GetService(serviceId, projectId, version string) (*model.Service, error) {
+	ctx := context.Background()
+	containers, err := d.client.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting service in docker unable to list containers got error message - %v", err)
+		return nil, err
+	}
+
+	service := new(model.Service)
+	service.ID = serviceId
+	service.ProjectID = projectId
+	service.Version = version
+	service.Whitelist = []string{fmt.Sprintf("%s:*", projectId)}
+	service.Upstreams = []model.Upstream{{ProjectID: projectId, Service: "*"}}
+	tasks := []model.Task{}
+	for _, containerInfo := range containers {
+		// while creating containers we assign single name to the container that is in the form < serviceId-taskId >
+		// so if length not equal to 1 throw error
+		if len(containerInfo.Names) != 1 {
+			logrus.Errorf("error getting service in docker containers length not equal to one")
+			return nil, err
+		}
+		containerName := containerInfo.Names[0]
+		// container name < serviceId-projectId-taskId-version >
+		if strings.Split(containerName, "-")[0] == serviceId {
+			containerInspect, err := d.client.ContainerInspect(ctx, containerName)
+			if err != nil {
+				logrus.Errorf("error getting service in docker unable to inspect container - %v", err)
+				return nil, err
+			}
+
+			service.Labels = containerInfo.Labels
+
+			// set ports of task
+			task := model.Task{}
+			ports := []model.Port{}
+			for portWithProtocol := range containerInspect.Config.ExposedPorts {
+				arr := strings.Split(string(portWithProtocol), "/")
+				portNumber, err := strconv.Atoi(arr[0])
+				if err != nil {
+					logrus.Errorf("error getting service in docker unable to convert string to int got error message - %v", err)
+					return nil, err
+				}
+				ports = append(ports, model.Port{Protocol: model.Protocol(arr[1]), Port: int32(portNumber)}) // port name remaining
+			}
+			task.Ports = ports
+
+			// set environment variable of task
+			envs := map[string]string{}
+			for _, value := range containerInspect.Config.Env {
+				env := strings.Split(value, "=")
+				envs[env[0]] = env[1]
+			}
+			delete(envs, "URL")
+			delete(envs, "TOKEN")
```

Comment:
Delete only if the service runtime was code. Store the runtime in a label as well